kinesis/kinesis.go

```diff
@@ -637,6 +647,81 @@ func gzipCompress(data []byte) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+// Compress Then Truncate
+// compresses data with CompressorFunction and iteratively truncates data
+// adding the truncation suffix if the CompressorFunction output exceeds maxOutLen.
+// The output is compressed and possibly truncated data whose length guaranteed to
+// be less than or equal to maxOutLen.
+func compressThenTruncate(compressorFunc CompressorFunc, data []byte, maxOutLen int, truncatedSuffix []byte, outputPlugin OutputPlugin) ([]byte, error) {
+	var compressedData []byte
+	var truncationBuffer []byte
+	var originalCompressedLen int
+	var compressedLen int
+	var err error
+
+	/* Iterative approach to truncation */
+	isTruncated := false
+	compressedLen = math.MaxInt64
+	truncatedInLen := len(data)
+	truncationBuffer = data
+	for (maxOutLen < compressedLen) {
```

Comment:
nit: it feels more logical to invert this as 

`compressedLen > maxOutLen`

Reply:
swapped.